src/main/java/com/artipie/rpm/Rpm.java

```diff
@@ -397,45 +398,24 @@ private Completable doWithLock(final Key target, final Supplier<Completable> ope
                                 .calculate(prefix)
                                 .thenApply(repo -> new PackagesDiff(primary, repo))
                         ).thenCompose(
-                            diff -> CompletableFuture.allOf(
-                                this.handlePackagesToRemove(prefix, diff.toDelete())
-                                    .toCompletableFuture(),
-                                Rpm.copyPackagesToAdd(
-                                    new SubStorage(prefix, this.storage),
-                                    diff.toAdd().keySet().stream().map(Key.From::new)
-                                        .collect(Collectors.toList())
-                                ).toCompletableFuture()
-                            )
+                            diff -> Rpm.copyPackagesToAdd(
+                                new SubStorage(prefix, this.storage),
+                                diff.toAdd().keySet().stream().map(Key.From::new)
+                                    .collect(Collectors.toList())
```

Comment:
I've changed `PackagesDiff.toAdd` signature to return collection instead of map. Mapping to `Key` instances I think is not a responsibly of  `PackagesDiff`, it should be done here.